tests/filecheck/transforms/acc-dedup.mlir

```diff
@@ -179,3 +179,65 @@ func.func @nested_loops(%A : i32, %lb : i32, %ub : i32, %step : i32) {
   func.return
 }
 
+// check that every loop-nest only contains values that cannot be set at a higher level
+// CHECK-NEXT:  func.func @nested_loops(%A_1 : i32, %lb_2 : i32, %ub_2 : i32, %step_2 : i32) {
+// CHECK-NEXT:    %21 = acc2.setup on "simple" ("A" = %A_1 : i32) : !acc2.state<"simple">
+//                                              ^^^^^^^^^^^^^^^^
+// CHECK-NEXT:    %22 = scf.for %i = %lb_2 to %ub_2 step %step_2 iter_args(%23 = %21) -> (!acc2.state<"simple">) : i32 {
+// CHECK-NEXT:      %24 = acc2.setup on "simple" ("i" = %i : i32) in_state(%23) : !acc2.state<"simple">
+//                                                ^^^^^^^^^^^^^^
+// CHECK-NEXT:      %25 = scf.for %j = %i to %ub_2 step %step_2 iter_args(%26 = %24) -> (!acc2.state<"simple">) : i32 {
+// CHECK-NEXT:        %out_state = acc2.setup on "simple" ("j" = %j : i32) in_state(%26) : !acc2.state<"simple">
+//                                                         ^^^^^^^^^^^^^^^
+// CHECK-NEXT:        scf.yield %out_state : !acc2.state<"simple">
+// CHECK-NEXT:      }
+// CHECK-NEXT:      scf.yield %25 : !acc2.state<"simple">
+// CHECK-NEXT:    }
+// CHECK-NEXT:    func.return
+// CHECK-NEXT:  }
+
+
+func.func @nested_loops_edge_cases(%A : i32, %lb : i32, %ub : i32, %step : i32) {
+  %0 = acc2.setup on "simple" () : !acc2.state<"simple">
+  %1 = scf.for %i = %lb to %ub step %step iter_args(%2 = %0) -> (!acc2.state<"simple">) : i32 {
+    %3 = scf.for %j = %i to %ub step %step iter_args(%l0 = %2) -> (!acc2.state<"simple">) : i32 {
+      %l1 = acc2.setup on "simple" ("A" = %A : i32, "B" = %A : i32, "i" = %i : i32, "j" = %j : i32) in_state(%l0) : !acc2.state<"simple">
+      // test.op is necessary, otherwise the two setups will be fused (as the second one would just overwrite the first one)
+      "test.op" () : () -> ()
```

Comment:
Can we insert a launchop/awaitop pair here instead?

Reply:
ok

Reply:
wait, adding a launch necessitates an await, which makes this test way to unreadable. I reverted back to setup ops as optimisation barriers, hope you are happy xx